src/mantarray_desktop_app/utils/generic.py

```diff
@@ -148,17 +148,39 @@ def convert_request_args_to_config_dict(request_args: Dict[str, Any]) -> Dict[st
     return config_dict
 
 
-def redact_sensitive_info_from_path(file_path: Optional[str]) -> Optional[str]:
-    """Scrubs username from file path to protect sensitive info."""
-    if file_path is None:
+def redact_sensitive_info(
```

Comment:
This isn't quite what we want, this method shouldn't be changed at all. We want a signle  method that you can pass an item into that will handle all possible redaction cases, whether it be a dict with a sensitive field, or a string with some sensitive info, let's just call it `redact_sensitive_info`. This method will call `redact_sensitive_info_from_path` when it encounters a file/folder path. We also need to move the loop further down back where it originally was in process monitor. Once the loop is moved back, you'll just be calling `redact_sensitive_info` on each item